### PR TITLE
Create design doc for cross-region S3 client

### PR DESCRIPTION
**Pull request by zoewangg**
## Motivation and Context
#52 

## Modifications
Create design doc for cross-region S3 client

### CODE REVIEW

File: docs/design/services/s3/cross-region-s3-client.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+// AWS CRT-based S3 async client
+S3AsyncClient s3CrtClient = S3AsyncClient.crtBuilder()
+                                         .crossRegionAcccessEnabled(true)
+                                         .build();
+```
+
+### Client Configuration
+
+Users can enable this feature through a client configuration on the client builder, `crossRegionAccessEnabled`. It is default to false.
```
Comment (L-Applin) [inline docs/design/services/s3/cross-region-s3-client.md:39]: small typo: `It is default to false`
should be `It is set to false by default` or something like that

File: docs/design/services/s3/cross-region-s3-client.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+## Specification 
+
+The SDK 2.x will support this feature in the S3 sync client, S3 async client and the AWS-CRT based S3 client.
+
+### Usage Examples
+
+#### Example 1: enabling cross-region access in the S3 sync client
+
+```
+S3ClientBuilder s3ClientBuilder = S3Client.builder()
```
Comment (joviegas) [inline docs/design/services/s3/cross-region-s3-client.md:18]: The `crossRegionAcccessEnabled` API will be made available to only "S3Client.builder"  and not to `AwsClientBuilder` which will make it available for all the clients even  if its not supported.
Thus we might need to add Customization for S3 only to show this API.

Comment (zoewangg) [reply docs/design/services/s3/cross-region-s3-client.md:18]: Yes, alternatively, we could consider adding it to `S3Configuration`, I don't have strong opinions. We still need to add `crossRegionAcccessEnabled` to the top level S3CrtAsyncClient builder though.

**Comment by zoewangg** [comment]
Closing because it's obsolete.

**Pull request closed without merging**